test/built-ins/RegExp/prototype/exec/failure-lastindex-access.js

```diff
@@ -18,16 +19,18 @@ info: >
               2. Return null.
 ---*/
 
-var thrower = {
+var gets = 0;
+var counter = {
   valueOf: function() {
-    throw new Test262Error();
+    gets++;
+    return 0;
   }
 };
 
 var r = /a/;
-r.lastIndex = thrower;
+r.lastIndex = counter;
 
 var result = r.exec('nbc');
 assert.sameValue(result, null);
-assert.sameValue(r.lastIndex, thrower);
-
+assert.sameValue(r.lastIndex, counter);
```

Comment:
nice approach, I would try some get/set accessor operation in there but this works and it's simple.